### ISSUE TITLE
Script crashes before menu loads

### ISSUE THREAD

**Issue by danfiscus** [datawhores/OF-Scraper#478]
[ofscraper_main_profile_2024-12-05.log](https://github.com/user-attachments/files/18032020/ofscraper_main_profile_2024-12-05.log)

## Describe the bug

I must be missing something here, but when I try to run the script, it doesn't present me a menu, it just crashes after showing the ascii art title. (See screenshot)

## To Reproduce

I installed the script through pip as described in the documentation.

## Expected behavior

I expected a menu of some kind to appear and present me with options.

## Screenshots/Logs

![image](https://github.com/user-attachments/assets/92beb266-5894-4510-ac38-511248a09fad)

## Config

Haven't touched the config file at all, should be whatever is generated/default. I did anonymize my local username in the uploaded config file below.
[config.json](https://github.com/user-attachments/files/18032077/config.json)

## System Info

- OS: Windows 10 Pro
- Browser: Vivaldi (Chromium-based)
- Version: Browser version 7.0.3495.15 (Stable channel) (64-bit) Script version latest (3.12.9)
- python
- installed via pip, 3.11.2

## Additional context

Not sure what else to add but happy to answer questions.

**Comment by cjb900** [comment]
This is a known issue. You need to make sure you have `aiolimiter` version 1.1.0 or lower

**Comment by danfiscus** [comment]
@cjb900 Thanks for clarifying, I did not see any mention of that in the documentation...

I've run `pip install --force-reinstall -v "aiolimiter==1.1.0"` and got a success message. When I re-run the `ofscraper` command, I'm having the same issue. What's the next step to debug this?

**Comment by cjb900** [comment]
I've only seen others recently having this issue with a version of `aiolimiter` higher than 1.1.0. So if you can confirm you have that version installed and you are indeed using the pip version of ofscraper, then something else in your install and or setup is causing that issue. 

You could try updating your version of Python to something like 3.11.6 (but not the latest 3.13 as it's known not to work) and see if that helps. If not then you would have to look through the logs to see what else may be causing your issue.

**Comment by danfiscus** [comment]
@cjb900 Okay, I also force reinstalled `ofscraper` through pip and the program seems to be running now. Must've been some combination of the `aiolimiter` version and a pip bug. I didn't see any open issues referencing aiolimiter so I'm assuming it's best to leave this open until the bug is resolved. I also just discovered a bug with the vivaldi auth fetch but I'm tired of debugging this so I'll let someone else open that issue.

**Comment by cjb900** [comment]
@danfiscus I just tested the latest version of Vivaldi and had no issues with fetching my auth information, adding it to ofscraper, and scraping content. So what ever issue you were encountering with Vivaldi I can't replicate. 

Vivaldi | 7.0.3495.23 (Stable channel) (64-bit)
Revision | 72d89227db29f524d89b1af38163f46e4ed7746e
OS | Windows 11 Version 23H2 (Build 22631.4541)
JavaScript | V8 13.0.245.20
User Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
Command Line | "C:\Users\cjb900\AppData\Local\Vivaldi\Application\vivaldi.exe" --flag-switches-begin --flag-switches-end --save-page-as-mhtml
Executable Path | C:\Users\cjb900\AppData\Local\Vivaldi\Application\vivaldi.exe
Profile Path | C:\Users\cjb900\AppData\Local\Vivaldi\User Data\Default

**Comment by JG20108** [comment]
Hi there! @cjb900 @danfiscus issue persists on my end. Things do not seem to work. Here is my output debug, any ideas?:


```
C:\Windows\System32>ofscraper --output debug
 [system.get_dupe_ofscraper:89]  Duplicated Processes []                                                   system.py:89
 [logs.print_system_log:52]  Log Level: DEBUG                                                                logs.py:52
 [logs.print_system_log:53]  version: 3.12.9                                                                 logs.py:53
 [logs.print_system_log:54]  Windows-11-10.0.26100-SP0                                                       logs.py:54
 [logs.print_system_log:55]  config path: C:\Users\your_username\.config\ofscraper\config.json               logs.py:55
 [logs.print_system_log:56]  profile path: C:\Users\your_username\.config\ofscraper\main_profile             logs.py:56
 [logs.print_system_log:57]  log folder: C:\Users\your_username\.config\ofscraper\logging                    logs.py:57
 [logs.print_system_log:58]  ssl DefaultVerifyPaths(cafile=None, capath=None,                                logs.py:58
openssl_cafile_env='SSL_CERT_FILE', openssl_cafile='C:\\Program Files\\Common Files\\SSL/cert.pem',
openssl_capath_env='SSL_CERT_DIR', openssl_capath='C:\\Program Files\\Common Files\\SSL/certs')
 [logs.print_system_log:59]  python version 3.12.0                                                           logs.py:59
 [logs.print_system_log:60]  certifi C:\Python312\Lib\site-packages\certifi\cacert.pem                       logs.py:60
 [logs.print_system_log:61]  number of threads available on system 24                                        logs.py:61
 [logs.print_args:66]  {'output': 'DEBUG', 'config': None, 'profile': None, 'no_rich': False, 'discord':     logs.py:66
'OFF', 'log': None, 'no_auto_resume': False, 'downloadbars': False, 'downloadsem': None, 'downloadthreads':
None, 'download_limit': None, 'quality': None, 'mediatype': [], 'size_max': None, 'size_min': None,
'length_max': None, 'length_min': None, 'protected_only': False, 'normal_only': False, 'media_id': [],
'original': False, 'space_replacer': None, 'textlength': None, 'text_type': 'word', 'posts': [],
'download_area': [], 'like_area': [], 'post_id': [], 'filter': ['.*'], 'neg_filter': [], 'scrape_paid':
False, 'block_ads': False, 'max_count': 0, 'media_sort': None, 'media_desc': False, 'post_sort': 'date',
'post_desc': False, 'redownload': False, 'force_all': False, 'force_model_unique': False, 'force_like':
False, 'label': [], 'download_text': [], 'before_original': None, 'before': <Arrow
[2025-02-08T13:33:56.466221-06:00]>, 'after': None, 'mass_msg': None, 'timed_only': None, 'timeline_strict':
False, 'action': [], 'daemon': None, 'usernames': [], 'excluded_username': [], 'user_list': [],
'black_list': [], 'current_price': None, 'renewal_price': None, 'regular_price': None, 'promo_price': None,
'last_seen': None, 'free_trail': None, 'promo': None, 'all_promo': None, 'sub_status': None, 'renewal':
None, 'promo_price_min': None, 'promo_price_max': None, 'regular_price_min': None, 'regular_price_max':
None, 'current_price_min': None, 'current_price_max': None, 'renewal_price_min': None, 'renewal_price_max':
None, 'last_seen_before': None, 'last_seen_after': None, 'expired_after': None, 'expired_before': None,
'subscribed_after': None, 'subscribed_before': None, 'sort': 'name', 'desc': False, 'users_first': False,
'individual': False, 'list': False, 'no_cache': False, 'no_api_cache': False, 'key_mode': None, 'keydb_api':
None, 'private_key': None, 'client_id': None, 'dynamic_rules': None, 'update_profile': False, 'auth_fail':
False, 'download_script': None, 'post_script': None, 'command': 'OF-Scraper'}
 [logs.print_args:67]  sys argv:['--output', 'debug']                                                        logs.py:67
 [logs.print_config:71]  {'main_profile': 'main_profile', 'metadata':                                        logs.py:71
'{configpath}/{profile}/.data/{model_id}', 'discord': '', 'file_options': {'save_location':
'C:\\Users\\your_username\\Data\\ofscraper', 'dir_format': '{model_username}/{responsetype}/{mediatype}/',
'file_format': '{filename}.{ext}', 'textlength': 0, 'space_replacer': ' ', 'date': 'MM-DD-YYYY',
'text_type_default': 'letter', 'truncation_default': True}, 'download_options': {'filter': ['Images',
'Audios', 'Videos'], 'auto_resume': True, 'system_free_min': 0, 'max_post_count': 0}, 'binary_options':
{'ffmpeg': ''}, 'cdm_options': {'private-key': None, 'client-id': None, 'key-mode-default': 'cdrm',
'keydb_api': ''}, 'performance_options': {'download_sems': 6, 'thread_count': 2, 'download_limit': 0},
'content_filter_options': {'block_ads': False, 'file_size_max': 0, 'file_size_min': 0, 'length_max': 0,
'length_min': 0}, 'advanced_options': {'code-execution': False, 'dynamic-mode-default': 'deviint',
'backend': 'aio', 'downloadbars': False, 'cache-mode': 'sqlite', 'appendlog': True, 'custom_values': None,
'sanitize_text': False, 'temp_dir': None, 'remove_hash_match': None, 'infinite_loop_action_mode': False,
'enable_auto_after': False, 'default_user_list': 'main', 'default_black_list': ''}, 'scripts_options':
{'post_download_script': None, 'post_script': None}, 'responsetype': {'timeline': 'Posts', 'message':
'Messages', 'archived': 'Archived', 'paid': 'Messages', 'stories': 'Stories', 'highlights': 'Stories',
'profile': 'Profile', 'pinned': 'Posts', 'streams': 'Streams'}, 'overwrites': {'audios': {}, 'videos': {},
'images': {}, 'text': {}}, 'ffmpeg': 'C:\\Users\\your_username\\.config\\ofscraper\\bin\\ffmpeg.exe'}
 [logs.print_start_message:82]  OF-Scraper Announcements:__(1). For those who use cdrm for cdm; a fix has    logs.py:82
been pushed to handle errors from the CDRM-Project API. Previously FFMPEG would endlessly fail for effected
downloads because of bad inputs being cached. Update to at least [3.12.7]__(2). cdrm seems to be getting ip
blocked by onlyfans based on error messages, the fix is to use your own keys or switch to keydb or to wait
for the requests to work again
 [logs.print_latest_version:94]  OF-Scraper up to date                                                       logs.py:94

    _______  _______         _______  _______  _______  _______  _______  _______  _______
    (  ___  )(  ____ \       (  ____ \(  ____ \(  ____ )(  ___  )(  ____ )(  ____ \(  ____ )
    | (   ) || (    \/       | (    \/| (    \/| (    )|| (   ) || (    )|| (    \/| (    )|
    | |   | || (__     _____ | (_____ | |      | (____)|| (___) || (____)|| (__    | (____)|
    | |   | ||  __)   (_____)(_____  )| |      |     __)|  ___  ||  _____)|  __)   |     __)
    | |   | || (                   ) || |      | (\ (   | (   ) || (      | (      | (\ (
    | (___) || )             /\____) || (____/\| ) \ \__| )   ( || )      | (____/\| ) \ \__
    (_______)|/              \_______)(_______/|/   \__/|/     \||/       (_______/|/   \__/



 [run.main:22]  Traceback (most recent call last):                                                            run.py:22
  File "C:\Python312\Lib\site-packages\ofscraper\runner\open\run.py", line 12, in main
    start_manager()
  File "C:\Python312\Lib\site-packages\ofscraper\runner\manager.py", line 17, in start_manager
    Manager.start()
  File "C:\Python312\Lib\site-packages\ofscraper\runner\manager.py", line 36, in start
    self.pick()
  File "C:\Python312\Lib\site-packages\ofscraper\runner\manager.py", line 44, in pick
    import ofscraper.commands.runners.check as check
  File "C:\Python312\Lib\site-packages\ofscraper\commands\runners\check.py", line 25, in <module>
    import ofscraper.actions.actions.download.normal.downloadnormal as downloadnormal
  File "C:\Python312\Lib\site-packages\ofscraper\actions\actions\download\normal\downloadnormal.py", line 23,
in <module>
    from ofscraper.classes.sessionmanager.download import download_session
  File "C:\Python312\Lib\site-packages\ofscraper\classes\sessionmanager\download.py", line 17, in <module>
    from ofscraper.classes.sessionmanager.leaky import LeakyBucket
  File "C:\Python312\Lib\site-packages\ofscraper\classes\sessionmanager\leaky.py", line 4, in <module>
    from aiolimiter.compat import wait_for
ModuleNotFoundError: No module named 'aiolimiter.compat'

 [run.main:23]  No module named 'aiolimiter.compat'                                                           run.py:23
 [close.sendCloseMessage:49]  Finished Script                                                               close.py:49`
```

**Comment by datawhores** [comment]
No response so closing for now 
Please make a new issue if needed
